awscli/customizations/s3/utils.py

```diff
@@ -449,50 +449,19 @@ def __init__(self, client, date_parser=_date_parser):
         self._date_parser = date_parser
 
     def list_objects(self, bucket, prefix=None, page_size=None):
-        kwargs = {'Bucket': bucket, 'EncodingType': 'url',
-                  'PaginationConfig': {'PageSize': page_size}}
+        kwargs = {'Bucket': bucket, 'PaginationConfig': {'PageSize': page_size}}
         if prefix is not None:
             kwargs['Prefix'] = prefix
-        # This event handler is needed because we use encoding_type url and
-        # we're paginating.  The pagination token is the last Key of the
-        # Contents list.  However, botocore does not know that the encoding
-        # type needs to be urldecoded.
-        with ScopedEventHandler(self._client.meta.events,
-                                'after-call.s3.ListObjects',
-                                self._decode_keys,
-                                'BucketListerDecodeKeys'):
-            paginator = self._client.get_paginator('list_objects')
-            pages = paginator.paginate(**kwargs)
-            for page in pages:
-                contents = page.get('Contents', [])
-                for content in contents:
-                    source_path = bucket + '/' + content['Key']
-                    content['LastModified'] = self._date_parser(
-                        content['LastModified'])
-                    yield source_path, content
-
-    def _decode_keys(self, parsed, **kwargs):
-        if 'Contents' in parsed:
-            for content in parsed['Contents']:
-                content['Key'] = unquote_str(content['Key'])
-
-
-class ScopedEventHandler(object):
```

Comment:
By the way, this is a really nice abstraction that I hadn't seen before. I wonder if it would be useful to customers that use botocore and boto3?